Cpp_test/test/pipe-test.cpp

```diff
@@ -0,0 +1,170 @@
+    ipc_info info = ipc.get_options();
+    IPCPipeSend pipe(info);
+    
+    int pd{0};
+	pd = open(info.method_name, O_RDONLY | O_NONBLOCK);
+
+	if (pd != -1){
+        EXPECT_NO_THROW(pipe.init());
+    }
+}
+
+TEST(PipeReceiveTest3, start){
+    optind =0;
+    char mode[] = {"./ipc_receive"};
+    char pipeMethod[] = {"-p"};
+    char methodName[] = {"methodName"};
+    char file[] = {"-f"};
+    char fileName[] = {"fileName"};
+    int argc = 5;
+    char* argv[] = {mode, pipeMethod, methodName, file, fileName};
+    ipcHandler ipc;
+    ipc_info info;
+    
+    if(info.mode == IPCMode::RECEIVE_MODE && info.protocol == IPCProtocol::PIPE){
+        ipc.select_options(IPCMode::RECEIVE_MODE,argc, argv);
+        ipc.get_options();
+        EXPECT_NO_THROW(ipc.start());
+    }
+}
+
+TEST(PipeSendTest4, initFailOpen){
+    optind =0;
+    char mode[] = {"./ipc_send"};
+    char pipeMethod[] = {"-p"};
+    char methodName[] = {"/methodName"};
+    char file[] = {"-f"};
+    char fileName[] = {"testFile/test"};
+    int argc = 5;
+    char* argv[] = {mode, pipeMethod, methodName, file, fileName};
+    ipcHandler ipc;
+    ipc.select_options(IPCMode::SEND_MODE,argc, argv);
+    ipc_info info = ipc.get_options();
+    IPCPipeSend pipe(info);
+    
+    int pd{0};
+	pd = open(info.method_name, O_RDONLY | O_NONBLOCK);
+
+	if (pd == -1){
+        EXPECT_THROW(pipe.init(), std::runtime_error);
+    }
+}
+
+TEST(PipeSendTest5, initSuccessOpen){
+    optind =0;
+    char mode[] = {"./ipc_send"};
+    char pipeMethod[] = {"-p"};
+    char methodName[] = {"methodName"};
+    char file[] = {"-f"};
+    char fileName[] = {"testFile/test"};
+    int argc = 5;
+    char* argv[] = {mode, pipeMethod, methodName, file, fileName};
+    ipcHandler ipc;
+    ipc.select_options(IPCMode::SEND_MODE,argc, argv);
+    ipc_info info = ipc.get_options();
+    IPCPipeSend pipe(info);
+    
+    int pd{0};
+	pd = open(info.method_name, O_RDONLY | O_NONBLOCK);
+
+	if (pd != -1){
+        EXPECT_NO_THROW(pipe.init());
+    }
+}
+
+TEST(PipeSendTest6, start){
+    optind =0;
+    char mode[] = {"./ipc_send"};
+    char pipeMethod[] = {"-p"};
+    char methodName[] = {"methodName"};
+    char file[] = {"-f"};
+    char fileName[] = {"testFile/test"};
+    int argc = 5;
+    char* argv[] = {mode, pipeMethod, methodName, file, fileName};
+    ipcHandler ipc;
+    ipc_info info;
+    FileHandler fileRead(fileName, FileMode::READ);
+    
+    if(info.mode == IPCMode::SEND_MODE && fileRead.get_file_size() == 0){
+        ipc.select_options(IPCMode::SEND_MODE,argc, argv);
+        ipc.get_options();
+        EXPECT_THROW(ipc.start(), std::runtime_error);
+    } else if(info.mode == IPCMode::SEND_MODE && info.protocol == IPCProtocol::PIPE){
+        ipc.select_options(IPCMode::SEND_MODE,argc, argv);
+        ipc.get_options();
+        EXPECT_NO_THROW(ipc.start());
+    }
+}
+
+TEST(pipeSendTest7, tranferFile){
+    optind =0;
+    boost::process::child c;
```

Comment:
As required in the story, you shall not use boost.

Reply:
I will change